pkg/events/sorting/sorting_test.go

```diff
@@ -273,13 +272,16 @@ func TestEventsChronologicalSorter_Start(t *testing.T) {
 	}
 }
 
-func retrieveEventsFromSorter(expectedEventsAmount int, sorterOutputChan chan *trace.Event, fatalErrorsChan chan error) ([]trace.Event, error) {
+func retrieveEventsFromSorter(expectedEventsAmount int, sorterOutputChan <-chan *trace.Event, fatalErrorsChan chan error) ([]trace.Event, error) {
 	ticker := time.NewTicker(time.Second)
 	outputList := make([]trace.Event, 0)
 	eventsReceived := 0
 	for {
 		select {
-		case event := <-sorterOutputChan:
+		case event, ok := <-sorterOutputChan:
```

Comment:
I think in the sorter itself we're not checking for the closed channel, but it's not relevant to fix the DATA RACE warnings.

Reply:
I will check it

Reply:
@AlonZivony any feedback here ?